jetty-ee9/jetty-ee9-webapp/src/test/java/org/eclipse/jetty/ee9/webapp/TempDirTest.java

```diff
@@ -140,42 +149,98 @@ public void jettyBaseWorkExists(WorkDir workDirExt) throws Exception
         Path workDir = jettyBase.resolve("work");
         FS.ensureDirExists(workDir);
         WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
-        Server server = new Server();
-        server.setTempDirectory(workDir.toFile());
+        _server = new Server();
+        _server.setTempDirectory(workDir.toFile());
         WebAppContext webAppContext = new WebAppContext();
-        server.setHandler(webAppContext);
+        _server.setHandler(webAppContext);
         webInfConfiguration.resolveTempDirectory(webAppContext);
         assertThat(webAppContext.getTempDirectory().getParentFile().toPath(), PathMatchers.isSame(workDir));
+        assertThat(webAppContext.getAttribute(ServletContext.TEMPDIR), is(webAppContext.getTempDirectory()));
     }
 
     @Test
     public void testTempDirDeleted(WorkDir workDir) throws Exception
     {
         // Create war on the fly
         Path testWebappDir = MavenTestingUtils.getProjectDirPath("src/test/webapp");
-        Path warFile = workDir.getEmptyPathDir().resolve("test.war");
-
-        Map<String, String> env = new HashMap<>();
-        env.put("create", "true");
-
-        URI uri = URI.create("jar:" + warFile.toUri().toASCIIString());
-        // Use ZipFS so that we can create paths that are just "/"
-        try (FileSystem zipfs = FileSystems.newFileSystem(uri, env))
-        {
-            Path root = zipfs.getPath("/");
-            IO.copyDir(testWebappDir, root);
-        }
 
         //Let jetty create the tmp dir on the fly
-        Server server = new Server();
+        _server = new Server();
         WebAppContext webAppContext = new WebAppContext();
         webAppContext.setContextPath("/");
-        webAppContext.setWarResource(webAppContext.getResourceFactory().newResource(warFile));
-        server.setHandler(webAppContext);
-        server.start();
+        webAppContext.setWar(testWebappDir.toFile().getAbsolutePath());
+        _server.setHandler(webAppContext);
+        _server.start();
         File tempDirectory = webAppContext.getTempDirectory();
-        server.stop();
+        assertThat(webAppContext.getAttribute(ServletContext.TEMPDIR), is(webAppContext.getTempDirectory()));
+        _server.stop();
         assertNull(webAppContext.getTempDirectory());
         assertThat("Temp dir exists", !Files.exists(tempDirectory.toPath()));
     }
+
+    @Test
+    public void testExplicitTempDir(WorkDir workDir) throws Exception
+    {
+        Path myTempDir = workDir.getEmptyPathDir().resolve("my-temp-dir");
+        FS.ensureDirExists(myTempDir);
+
+        // Create war on the fly
+        Path testWebappDir = MavenTestingUtils.getProjectDirPath("src/test/webapp");
```

Comment:
this will not work with generated ee8 as resources are in target/test-classes.
Need to be (but depends if the test only need the web.xml or the other resources at the web.xml will not be in this directory)
```suggestion
        Path testWebappDir = MavenTestingUtils.getTargetPath("test-classes/webapp");
```

Reply:
The new `MavenPaths` object has several `.find*` methods that will work as intended on ee8/ee9.
Can we use one of those?